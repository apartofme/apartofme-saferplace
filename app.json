{
  "name": "SaferPlace",
  "displayName": "Safer Place"
}
