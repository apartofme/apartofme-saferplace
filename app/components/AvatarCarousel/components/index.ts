export * from './AvatarCarouselItem';
export * from './ProgressBarItem';
