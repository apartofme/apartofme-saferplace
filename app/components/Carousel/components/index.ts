export * from './SubtitleImage/SubtitleImage';
export * from './ImageTitleSubTitle';
export * from './ImageSubtitle';
export * from './OnlyImage';
export * from './ProgressBarItem';
