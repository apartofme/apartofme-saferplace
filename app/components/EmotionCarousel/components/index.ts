export * from './EmotionCarouselItem';
export * from './ProgressBarItem';
