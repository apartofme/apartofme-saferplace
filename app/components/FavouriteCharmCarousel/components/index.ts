export * from './FavouriteCharmCarouselItem';
export * from './ProgressBarItem';
