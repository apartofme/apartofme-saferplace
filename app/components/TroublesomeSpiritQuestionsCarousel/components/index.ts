export * from './TroublesomeSpiritQuestionsCarouselItem';
export * from './ProgressBarItem';
