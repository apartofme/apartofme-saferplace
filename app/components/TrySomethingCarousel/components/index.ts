export * from './TrySomethingCarouselItem';
export * from './ProgressBarItem';
