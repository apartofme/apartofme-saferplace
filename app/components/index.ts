export * from './ExtendedButton';
export * from './ExtendedText';
export * from './ExtendedTextInput';
