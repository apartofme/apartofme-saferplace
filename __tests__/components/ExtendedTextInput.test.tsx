import 'react-native';
import React from 'react';
import testRenderer from 'react-test-renderer';

import { ExtendedTextInput, ExtendedTextInputType } from '../../app/components';

test('UI: ExtendedTextInput should match snapshot', () => {
  // Default ExtendedTextInput case
  const actualExtendedTextInput = testRenderer
    .create(
      <ExtendedTextInput
        label="test input"
        labelStyle={{ color: 'red', marginBottom: 10 }}
        type={ExtendedTextInputType.Default}
        value="test input"
        onChangeText={() => {}}
        error="smth broken"
        errorStyle={{ color: 'red', marginTop: 10 }}
        style={{ backgroundColor: 'yellow' }}
      />,
    )
    .toJSON();
  expect(actualExtendedTextInput).toMatchSnapshot();

  // Specific ExtendedTextInput type case
  const actualExtendedPasswordToggleTextInput = testRenderer
    .create(
      <ExtendedTextInput
        label="password toggle test input"
        labelStyle={{ color: 'red', marginBottom: 10 }}
        type={ExtendedTextInputType.PasswordToggle}
        value="test input"
        onChangeText={() => {}}
        error="smth broken"
        errorStyle={{ color: 'red', marginTop: 10 }}
        style={{ backgroundColor: 'yellow' }}
      />,
    )
    .toJSON();
  expect(actualExtendedPasswordToggleTextInput).toMatchSnapshot();
});
