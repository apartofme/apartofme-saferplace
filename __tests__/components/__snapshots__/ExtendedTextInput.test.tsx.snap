// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI: ExtendedTextInput should match snapshot 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 15,
        },
        Array [
          Object {},
          Object {
            "color": "red",
            "marginBottom": 10,
          },
        ],
      ]
    }
  >
    test input
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
        },
        Object {
          "backgroundColor": "yellow",
        },
      ]
    }
    value="test input"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 15,
        },
        Array [
          Object {},
          Object {
            "color": "red",
            "marginTop": 10,
          },
        ],
      ]
    }
  >
    smth broken
  </Text>
</View>
`;

exports[`UI: ExtendedTextInput should match snapshot 2`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 15,
        },
        Array [
          Object {},
          Object {
            "color": "red",
            "marginBottom": 10,
          },
        ],
      ]
    }
  >
    password toggle test input
  </Text>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Array [
          Array [
            Object {
              "borderBottomWidth": 1,
            },
            Object {
              "backgroundColor": "yellow",
            },
          ],
          Object {
            "borderBottomWidth": 0,
            "flex": 1,
          },
        ]
      }
      value="test input"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
            "height": 15,
            "width": 15,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 15,
        },
        Array [
          Object {},
          Object {
            "color": "red",
            "marginTop": 10,
          },
        ],
      ]
    }
  >
    smth broken
  </Text>
</View>
`;
